libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -166,3 +166,81 @@ string IRStorageArrayLength::setToZero() const
 {
 	return storeValue("0", *TypeProvider::uint256());
 }
+
+IRMemoryItem::IRMemoryItem(
+	IRGenerationContext& _context,
+	std::string _address,
+	bool _byteArrayElement,
+	Type const& _type
+):
+	IRLValue(_context, &_type),
+	m_address(move(_address)),
+	m_byteArrayElement(_byteArrayElement)
+{ }
+
+string IRMemoryItem::retrieveValue() const
+{
+	solUnimplementedAssert(!m_byteArrayElement, "Byte arrays not yet implemented!");
+
+	if (m_type->isValueType())
+		return m_context.utils().readFromMemory(*m_type) +
+			"(" +
+			m_address +
+			")\n";
+	else
+		return "mload(" + m_address + ")\n";
+}
+
+string IRMemoryItem::storeValue(string const& _value, Type const& _type) const
+{
+	if (!m_type->isValueType())
+	{
+		solUnimplementedAssert(_type == *m_type, "Conversion not implemented for assignment to memory.");
+
+		solAssert(m_type->sizeOnStack() == 1, "");
+		solAssert(dynamic_cast<ReferenceType const*>(m_type), "");
+
+		return "mstore(" + m_address + ", " + _value + ")\n";
+	}
+
+	solAssert(_type.isValueType(), "");
+
+	string prepared = _value;
+
+	// Exists to see if this case ever happens
+	solAssert(_type == *m_type, "");
+
+	if (_type != *m_type)
+		prepared =
+			m_context.utils().conversionFunction(_type, *m_type) +
+			"(" +
+			_value +
+			")";
+	else
+		prepared =
+			m_context.utils().cleanupFunction(*m_type) +
+			"(" +
+			_value +
+			")";
+
+	if (m_byteArrayElement)
+	{
+		solAssert(m_type->calldataEncodedSize(false) == 1, "Invalid non-padded type.");
```

Comment:
Please assert `m_type == *TypeProvider::byte()` instead and remove the next two lines

Reply:
Ah sorry, we still need the next two lines!

Reply:
Or at least the `byte` call.

Reply:
so.. remove the `FixedBytes` check and always do the byte call?

Reply:
If you remove the check (but not the if body) in https://github.com/ethereum/solidity/pull/7015/files#r299991316 then it's fine.

Reply:
sorry, your link probably broke after I pushed again.. could you link again / describe it?

Reply:
or maybe it's fine now?

Reply:
Yep, fine now!